linera-views/src/backends/scylla_db.rs

```diff
@@ -324,54 +429,143 @@ impl ScyllaDbClient {
         // Read the value of a key
         let values = (root_key.to_vec(), key);
 
-        let results = session
-            .execute_unpaged(&self.contains_key, &values)
-            .await?
-            .into_rows_result()?;
-        let mut rows = results.rows::<(Vec<u8>,)>()?;
+        let (result, _) = session
+            .execute_single_page(&self.contains_key, &values, PagingState::start())
+            .await?;
+        let rows = result.into_rows_result()?;
+        let mut rows = rows.rows::<(Vec<u8>,)>()?;
         Ok(rows.next().is_some())
     }
 
+    fn attempt_sticky_shard_policy_creation(
+        session: &Session,
+        namespace: &str,
+        partition_key: &[u8],
+        token: Option<Token>,
+        cache: &mut LruCache<Vec<u8>, LoadBalancingPolicyCacheEntry>,
+    ) -> Arc<dyn LoadBalancingPolicy> {
+        match StickyShardPolicy::new(
+            session,
+            namespace,
+            partition_key,
+            token,
+            ScyllaDbClient::build_default_policy(),
+        ) {
+            Ok(policy) => {
+                let policy = Arc::new(policy);
+                cache.push(
+                    partition_key.to_vec(),
+                    LoadBalancingPolicyCacheEntry::Ready(policy.clone()),
+                );
+                policy
+            }
+            Err(error) => {
+                // Cache that the policy creation failed, so we don't try again too soon, and don't
+                // recalculate the token if not needed.
+                let token = match error {
+                    ScyllaDbStoreInternalError::MissingTokenEndpoints(token) => Some(token),
+                    _ => None,
+                };
+                cache.push(
+                    partition_key.to_vec(),
+                    LoadBalancingPolicyCacheEntry::NotReady(Timestamp::now(), token),
+                );
+                ScyllaDbClient::build_default_policy()
+            }
+        }
+    }
+
+    // Returns a batch query with a sticky shard policy, that always tries to route to the same
+    // ScyllaDB shard.
+    // Should be used only on batches where all statements are to the same partition key.
+    async fn get_sticky_batch_query(
```

Comment:
What's happening here? Do we need this? This looks pretty non-trivial and the PR summary doesn't mention it.